Automattic/wp-calypso
# Adds Scan upsell when not available

#### Changes proposed in this Pull Request

* Adds a new scan page to promote Scan if it's not part of the site purchases.

#### Implementation notes

There's a few things that could be improved here:

* The detection on whether scan is enabled for a given site is entirely based around the site purchases.
* The purchases _only_ look for the Scan product, it doesn't check plans that would contain it.
* Enabling redirects to properly send users to a given destination to purchase Scan are needed.
* We still need to properly set the sidebar threat indicator, but I didn't want to bloat the PR, especially if we're improving the scan detection.

#### Testing instructions



* Load Jetpack Cloud.
* Select a site without scan purchased.
* Visit scan and scan history.
* Select a site with scan purchased.
* Visit scan and scan history.

Expected: sites without scan show new upsell screen, below. Sites with scan function as before.

<img width="1146" alt="Screen Shot 2020-04-08 at 4 28 24 PM" src="https://user-images.githubusercontent.com/1760168/78830483-0732cd00-79b6-11ea-9165-24dcf54f1828.png">

Fixes 1151678672052943-as-1170324772452809.
